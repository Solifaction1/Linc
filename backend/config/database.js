const path = require('path');

const connection = process.env.NODE_ENV !== 'production' ?
  {
    client: 'sqlite',
    connection: {
      filename: path.join(__dirname, '..', '.tmp/data.db'),
    },
    useNullAsDefault: true,
  } : {
    client: "postgres",
    acquireConnectionTimeout: 120000,
    connection: {
      // host: `/cloudsql/${process.env.INSTANCE_CONNECTION_NAME}`, // for a PostgreSQL database
      host: process.env.DATABASE_HOST, // for a PostgreSQL database
      database: process.env.DATABASE_NAME,
      user: process.env.DATABASE_USERNAME,
      password: process.env.DATABASE_PASSWORD,
      ssl: false,

    },
    useNullAsDefault: true,
    debug: false,
  };

module.exports = ({ }) => ({
  connection: connection,
});
