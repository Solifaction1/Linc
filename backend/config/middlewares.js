module.exports = [
  'strapi::errors',
  {
    name: 'strapi::cors',
    config: {
      enabled: true,
      headers: ['*'],
      origin: ['*']
    }
  },
  'strapi::poweredBy',
  'strapi::logger',
  'strapi::query',
  'strapi::body',
  'strapi::session',
  'strapi::favicon',
  'strapi::public',
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'connect-src': ["'self'", 'https:'],
          'img-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            "cdn.jsdelivr.net",
            "strapi.io",
            "s3.amazonaws.com",
            "editor.unlayer.com",
            process.env.CDN_URL,
          ],
          "script-src": ["'self'", "editor.unlayer.com"],
          "frame-src": ["'self'", "editor.unlayer.com"],
          'media-src': [
            "'self'",
            'data:',
            'blob:',
            'dl.airtable.com',
            process.env.CDN_URL,
          ],
          upgradeInsecureRequests: null,
        },
      },
    },
  },
];
