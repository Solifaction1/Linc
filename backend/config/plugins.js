module.exports = ({ env }) => ({
  "strapi-plugin-populate-deep": {
    config: {
      defaultDepth: 5, // Default is 5
    },
  },
  "duplicate-button": true,
  email: {
    config: {
      provider: "mailgun",
      providerOptions: {
        key: env("MAILGUN_API_KEY"), // Required
        domain: env("MAILGUN_DOMAIN"), // Required
        url: "https://api.eu.mailgun.net", //Optional. If domain region is Europe use 'https://api.eu.mailgun.net'
      },
      settings: {
        defaultFrom: "<mg@linc.legal>",
        defaultReplyTo: "<mg@linc.legal>",
      },
    },
  },
  upload: {
    config: {
      provider: "aws-s3",
      providerOptions: {
        accessKeyId: env("AWS_ACCESS_KEY_ID"),
        secretAccessKey: env("AWS_ACCESS_SECRET"),
        region: env("AWS_REGION"),
        params: {
          Bucket: env("AWS_BUCKET"),
        },
      },
      actionOptions: {
        upload: {},
        uploadStream: {},
        delete: {},
      },
    },
  },
  "content-versioning": {
    enabled: false,
  },
  slugify: {
    enabled: false,
    config: {
      contentTypes: {
        blog: {
          field: "slug",
          references: "title",
        },
      },
    },
  },
  "preview-button": {
    enabled: true,
    resolve: "./src/plugins/preview-button", // path to plugin folder
  },
  "email-designer": {
    enabled: true,

    // ⬇︎ Add the config property
    config: {
      editor: {
        // optional - if you have a premium unlayer account
        // projectId: [UNLAYER_PROJECT_ID],

        tools: {
          heading: {
            properties: {
              text: {
                value: "This is the new default text!",
              },
            },
          },
        },
        options: {
          features: {
            colorPicker: {
              presets: ["#D9E3F0", "#F47373", "#697689", "#37D67A"],
            },
          },
          fonts: {
            showDefaultFonts: false,
            /*
             * If you want use a custom font you need a premium unlayer account and pass a projectId number :-(
             */
            customFonts: [
              {
                label: "Anton",
                value: "'Anton', sans-serif",
                url: "https://fonts.googleapis.com/css?family=Anton",
              },
              {
                label: "Lato",
                value: "'Lato', Tahoma, Verdana, sans-serif",
                url: "https://fonts.googleapis.com/css?family=Lato",
              },
              // ...
            ],
          },
          mergeTags: [
            {
              name: "Naam",
              value: "{{= USER.Naam }}",
              sample: "Naam",
            },
            {
              name: "Voornaam",
              value: "{{= USER.Voornaam }}",
              sample: "Voornaam",
            },
            {
              name: "Email",
              value: "{{= USER.Email }}",
              sample: "john@doe.com",
            },
            {
              name: "Telefoonnummer",
              value: "{{= USER.Telefoonnummer }}",
              sample: "0123456789",
            },
            {
              name: "select",
              value: "{{= USER.select }}",
              sample: "select",
            },
            {
              name: "textArea",
              value: "{{= USER.textArea }}",
              sample: "textArea",
            },
            {
              name: "Form ID",
              value: "{{= USER.formId }}",
              sample: "0",
            },
          ],
        },
        appearance: {
          theme: "dark",
          panels: {
            tools: {
              dock: "left",
            },
          },
        },
      },
    },
  },
  seo: {
    enabled: true,
  },
  ckeditor: {
    enabled: true,
    config: {
      plugin: {
        // disable data-theme tag setting //
        // setAttribute:false,
        // disable strapi theme, will use default ckeditor theme //
        // strapiTheme:false,
        // styles applied to editor container (global scope) //
        // styles:`
        // :root{
        //   --ck-color-focus-border:red;
        //   --ck-color-text:red;
        // }
        // `
        styles: `
        .ck-editor__main p,
        .ck.ck-list__item .ck-button .ck-button__label {
          line-height: normal;
        }`,
      },
      editor: {
        // editor default config

        // https://ckeditor.com/docs/ckeditor5/latest/features/markdown.html
        // if you need markdown support and output set: removePlugins: [''],
        // default is
        removePlugins: ["Markdown"],

        // https://ckeditor.com/docs/ckeditor5/latest/features/toolbar/toolbar.html
        toolbar: {
          items: [
            "paragraph",
            "heading1",
            "heading2",
            "heading3",
            "|",
            "bold",
            "italic",
            "underline",
            "fontSize",
            "removeFormat",
            "|",
            "bulletedList",
            "numberedList",
            "alignment",
            "|",
            "StrapiMediaLib",
            "link",
            "horizontalLine",
            "|",
            "sourceEditing",
            "fullScreen",
            "undo",
            "redo",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/font.html
        fontSize: {
          options: [
            9,
            11,
            13,
            {
              title: "default",
              model: "18px",
            },
            17,
            19,
            21,
            27,
            35,
          ],
          supportAllValues: false,
        },
        fontFamily: {
          options: [
            "default",
            "Arial, Helvetica Neue, Helvetica, Source Sans Pro, sans-serif",
            "Courier New, Courier, monospace",
            "Georgia, serif",
            "Lucida Sans Unicode, Lucida Grande, sans-serif",
            "Tahoma, Geneva, sans-serif",
            "Times New Roman, Times, serif",
            "Trebuchet MS, Helvetica, sans-serif",
            "Verdana, Geneva, sans-serif",
            "Roboto, Roboto Black, Roboto Medium, Roboto Light, sans-serif",
          ],
          supportAllValues: true,
        },
        fontColor: {
          columns: 5,
          documentColors: 10,
        },
        fontBackgroundColor: {
          columns: 5,
          documentColors: 10,
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
        // default language: 'en',
        // https://ckeditor.com/docs/ckeditor5/latest/features/images/images-overview.html
        image: {
          resizeUnit: "%",
          resizeOptions: [
            {
              name: "resizeImage:original",
              value: null,
              icon: "original",
            },
            {
              name: "resizeImage:25",
              value: "25",
              icon: "small",
            },
            {
              name: "resizeImage:50",
              value: "50",
              icon: "medium",
            },
            {
              name: "resizeImage:75",
              value: "75",
              icon: "large",
            },
          ],
          toolbar: [
            "toggleImageCaption",
            "imageTextAlternative",
            "imageStyle:inline",
            "imageStyle:block",
            "imageStyle:side",
            "linkImage",
            "resizeImage:25",
            "resizeImage:50",
            "resizeImage:75",
            "resizeImage:original",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/table.html
        table: {
          contentToolbar: [
            "tableColumn",
            "tableRow",
            "mergeTableCells",
            "tableCellProperties",
            "tableProperties",
            "toggleTableCaption",
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/headings.html
        heading: {
          options: [
            {
              model: "paragraph",
              title: "Paragraph",
              class: "ck-heading_paragraph",
            },
            {
              model: "heading1",
              view: "h1",
              title: "Heading 1",
              class: "ck-heading_heading1",
            },
            {
              model: "heading2",
              view: "h2",
              title: "Heading 2",
              class: "ck-heading_heading2",
            },
            {
              model: "heading3",
              view: "h3",
              title: "Heading 3",
              class: "ck-heading_heading3",
            },
          ],
        },
        // https://ckeditor.com/docs/ckeditor5/latest/features/general-html-support.html
        htmlSupport: {
          allow: [
            {
              name: "img",
              attributes: {
                sizes: true,
                loading: true,
              },
            },
          ],
        },
      },
    },
  },
});
