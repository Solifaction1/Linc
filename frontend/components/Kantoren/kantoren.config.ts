import KantorenData from '@models/kantoren.model'

const defaultProps: KantorenData = {
	title: 'kantoren 1',
	description: {
		text: '<p>description</p>',
		className: 'string',
	},
	kantoorCollection: {
		data: [
			{
				id: 1,
				attributes: {
					title: 'Kantoor 1',
					phone: '034792341',
					adress: 'teststraat 2000 Antwerpen',
					email: 'test@email.be',
					slug: 'slug-1',
					image: {
						data: null,
					},
					component: [],
					locale: 'locale',
					seo: null,
					createdAt: '',
					updatedAt: '',
					publishedAt: '',
					body: [],
				},
			},
			{
				id: 2,
				attributes: {
					title: 'Kantoor 2 met een iets langere titel',
					phone: '037293456',
					adress: 'blastraat 2000 Antwerpen',
					email: 'bla@email.be',
					slug: 'slug-2',
					image: {
						data: null,
					},
					component: [],
					locale: 'locale',
					seo: null,
					createdAt: '',
					updatedAt: '',
					publishedAt: '',
					body: [],
				},
			},
		],
	},
	ctaLabel: 'ctaLabel'
}

export { defaultProps }
