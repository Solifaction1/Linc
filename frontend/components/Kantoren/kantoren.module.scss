@import 'styles/vars';
@import 'styles/utilities';

.kantoor {
	@extend %page-container;
	grid-template-areas:
		'kantoorInformation kantoorInformation kantoorInformation kantoorInformation kantoorInformation kantoorInformation kantoorInformation kantoorInformation kantoorInformation kantoorInformation kantoorInformation kantoorInformation'
		'kantoren kantoren kantoren kantoren kantoren kantoren kantoren kantoren kantoren kantoren kantoren kantoren';
	padding-top: 0;

	.kantoorInformation {
		grid-area: kantoorInformation;

		h2 {
			@extend %h2;
			margin-bottom: 2rem;
			color: $color-beige-dark;
			text-align: center;
			@media (max-width: $break-lg) {
				text-align: left;
			}
		}

		.text p {
			@extend %body-copy-1;
			text-align: center;
			@media (max-width: $break-lg) {
				text-align: left;
			}
		}
	}

	.kantorenContainer {
		grid-area: kantoren;

		.kantorenWrapper {
			display: flex;
			align-items: flex-start;
			@media (max-width: $break-lg) {
				flex-wrap: wrap;
				align-items: unset;
				justify-content: center;
			}

			.kantoor {
				display: flex;
				flex-direction: column;

				max-width: 26.9rem;
				margin-top: 10rem;
				padding: unset;

				@media (max-width: $break-lg) {
					max-width: 50%;
					margin-left: 0;
				}

				@media (max-width: $break-md) {
					max-width: 100%;
				}

				.title {
					position: relative;

					max-width: 17.2rem;
					min-height: 9rem;
					margin-bottom: 2rem;
					padding-bottom: 2rem;

					color: $color-beige-dark;

					font-size: $font-size-title-md-m;
					line-height: calc(
						#{$font-size-title-md-m} * #{$line-height-multiplier-title}
					);

					&::after {
						content: '';
						position: absolute;
						bottom: 0;
						left: 0;

						width: 100%;
						height: 1px;

						background: $color-beige-dark;
					}

					@media (max-width: $break-lg) {
						max-width: 80%;
					}

					@media (max-width: $break-md) {
						max-width: 100%;
					}
				}

				.phone,
				.adress {
					@extend %body-copy-1;
					color: gray;
				}

				.button {
					width: fit-content;
					margin-top: 4rem;
				}
			}
		}
	}
}
