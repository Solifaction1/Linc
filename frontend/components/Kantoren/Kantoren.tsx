import React from 'react'
import sanitizeHtml from 'sanitize-html'
import KantorenData from '@models/kantoren.model'
import YGWYS from '@components/YGWYS/YGWYS'
import Button from '@components/Button/Button'
import { defaultProps } from './kantoren.config'
import styles from './kantoren.module.scss'

const Kantoren = (props: KantorenData) => {
	return (
		<section className={styles.kantoor}>
			<div className={styles.kantoorInformation}>
				{props.title && <h2>{props.title}</h2>}
				{props.description?.text && (
					<div className={styles.text}>
						<YGWYS text={props.description?.text} />
					</div>
				)}
			</div>

			<div className={styles.kantorenContainer}>
				<div className={styles.kantorenWrapper}>
					{props.kantoorCollection?.data.map((kantoor) => (
						<div className={styles.kantoor} key={kantoor.id}>
							<p className={styles.title}>{kantoor.attributes.title}</p>
							{kantoor.attributes.adress && (
								<p
									className={styles.adress}
									dangerouslySetInnerHTML={{
										__html: sanitizeHtml(
											JSON.stringify(kantoor.attributes.adress)
												.replace(/\\n/g, '<br/>')
												.replace(/"([^"]*)"/g, '$1')
										),
									}}
								/>
							)}
							{kantoor.attributes.phone && (
								<a
									className={styles.phone}
									href={`tel:${kantoor.attributes.phone}`}
								>
									Tel.: {kantoor.attributes.phone}
								</a>
							)}
							<br />
							{kantoor.attributes.email && props.ctaLabel && (
								<Button
									label={props.ctaLabel}
									className={styles.button}
									handleClick={() => {
										window.open(`mailto:${kantoor.attributes.email}`)
									}}
								/>
							)}
						</div>
					))}
				</div>
			</div>
		</section>
	)
}

Kantoren.defaultProps = defaultProps

export default Kantoren
