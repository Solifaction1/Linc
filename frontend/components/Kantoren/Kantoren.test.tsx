import React from 'react'

import { findByText, render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { defaultProps } from './kantoren.config'
import Kantoren from './Kantoren'

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}))

describe('RechtsdomeinenCTA', () => {
	it('renders a RechtsdomeinenCTA', () => {
		const { container, getByText, getAllByText } = render(
			<Kantoren {...defaultProps} />
		)

		expect(container).toMatchSnapshot()

		expect(container.querySelector('section')).toBeVisible()
		expect(container.getElementsByClassName('kantoor')[0]).toBeVisible()
		expect(
			container.getElementsByClassName('kantoorInformation')[0]
		).toBeVisible()

		expect(container.getElementsByClassName('title')[0]).toBeVisible()
		expect(getByText('kantoren 1', { selector: 'h2' })).toBeVisible()

		expect(container.getElementsByClassName('text')[0]).toBeVisible()
		expect(getByText('description', { selector: 'p' })).toBeVisible()

		expect(
			container.getElementsByClassName('kantorenContainer')[0]
		).toBeVisible()
		expect(container.getElementsByClassName('kantorenWrapper')[0]).toBeVisible()

		expect(container.getElementsByClassName('kantoor')[0]).toBeVisible()
		expect(getByText('Kantoor 1', { selector: 'p' })).toBeVisible()
		expect(getByText('Tel.: 034792341', { selector: 'a' })).toBeVisible()
		expect(
			getByText('teststraat 2000 Antwerpen', { selector: 'p' })
		).toBeVisible()
		expect(getAllByText('ctaLabel', { selector: 'p' })[0]).toBeVisible()

		expect(container.getElementsByClassName('kantoor')[1]).toBeVisible()
		expect(
			getByText('Kantoor 2 met een iets langere titel', { selector: 'p' })
		).toBeVisible()
		expect(getByText('Tel.: 037293456', { selector: 'a' })).toBeVisible()
		expect(
			getByText('blastraat 2000 Antwerpen', { selector: 'p' })
		).toBeVisible()

		expect(getAllByText('ctaLabel', { selector: 'p' })[1]).toBeVisible()
	})
})
