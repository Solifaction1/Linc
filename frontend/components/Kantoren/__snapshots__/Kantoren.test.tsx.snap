// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RechtsdomeinenCTA renders a RechtsdomeinenCTA 1`] = `
<div>
  <section
    class="kantoor"
  >
    <div
      class="kantoorInformation"
    >
      <h2>
        kantoren 1
      </h2>
      <div
        class="text"
      >
        <div
          class="ygwys 
			undefined undefined"
        >
          <p>
            description
          </p>
        </div>
      </div>
    </div>
    <div
      class="kantorenContainer"
    >
      <div
        class="kantorenWrapper"
      >
        <div
          class="kantoor"
        >
          <p
            class="title"
          >
            Kantoor 1
          </p>
          <p
            class="adress"
          >
            teststraat 2000 Antwerpen
          </p>
          <a
            class="phone"
            href="tel:034792341"
          >
            Tel.: 
            034792341
          </a>
          <br />
          <button
            class="button false primary md false button beige"
            type="button"
          >
            <p
              class="label"
            >
              ctaLabel
            </p>
          </button>
        </div>
        <div
          class="kantoor"
        >
          <p
            class="title"
          >
            Kantoor 2 met een iets langere titel
          </p>
          <p
            class="adress"
          >
            blastraat 2000 Antwerpen
          </p>
          <a
            class="phone"
            href="tel:037293456"
          >
            Tel.: 
            037293456
          </a>
          <br />
          <button
            class="button false primary md false button beige"
            type="button"
          >
            <p
              class="label"
            >
              ctaLabel
            </p>
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
